log4j-core/src/main/java/org/apache/logging/log4j/core/config/plugins/processor/PluginProcessor.java

```diff
@@ -39,14 +41,20 @@
 import javax.lang.model.SourceVersion;
 import javax.lang.model.element.Element;
 import javax.lang.model.element.ElementVisitor;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
 import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
 import javax.lang.model.util.Elements;
 import javax.lang.model.util.SimpleElementVisitor7;
+import javax.lang.model.util.Types;
 import javax.tools.Diagnostic;
 import javax.tools.FileObject;
 import javax.tools.StandardLocation;
+import org.apache.commons.lang3.StringUtils;
```

Comment:
This is the reason the OSGi builds fail.

We don't have `commons-lang3` as direct dependency. It is brought in the project by the **optional** `commons-compress` dependency.

BND does not automatically mark optional Maven dependencies as optional OSGi dependencies and for good reasons (see bndtools/bnd#2713 for a discussion), so the OSGi tests detected that the **required** `org.apache.commons.lang3` package is missing. Please don't use Commons Lang 3, we are a bottom-of-the-stack library with no required dependencies.

Reply:
I see, yeah log4j is pretty bottom stack, we should be minimizing dependencies, thanks a lot @ppkarwasz for finding this out! I am sorry that you had to struggle for this. I have corrected it now :)